server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -152,4 +138,59 @@ private String getVideoId(String response) {
     // Only returning the first for now, potential to change if I optimize search results.
     return searchResults.get(0);
   }
+
+  /**
+   * Parses the comment JSON string and retrieves the comments.
+   *
+   * @param response The JSON string to be parsed.
+   * @return An ArrayList with each comment
+   */
+  private ArrayList<String> retrieveComments(String response) {
+    ArrayList<String> commentList = new ArrayList<>();
+
+    JsonElement jElement = JsonParser.parseString(response);
+    JsonObject jObject = jElement.getAsJsonObject();
+    JsonArray itemsArray = jObject.getAsJsonArray("items");
+
+    for (JsonElement el : itemsArray) {
+      String topComment =
+          el.getAsJsonObject()
+              .getAsJsonObject("snippet")
+              .getAsJsonObject("topLevelComment")
+              .getAsJsonObject("snippet")
+              .get("textOriginal")
+              .toString();
+
+      // Remove all non-ASCII characters
+      commentList.add(topComment.replaceAll("[^\\x00-\\x7F]", ""));
+    }
+
+    return commentList;
+  }
+
+  /**
+   * Condenses array of comments into one large string, formatting it along the way and separating
+   * unpunctuated sentences with a period
+   *
+   * @param comments The array to be condensed.
+   * @return A properly formatted String
+   */
+  private String convertToString(ArrayList<String> comments) {
+    StringBuilder res = new StringBuilder();
+
+    for (String string : comments) {
```

Comment:
nit: consider renaming "string" to "comment" to improve readability in this block